### DISCUSSION TITLE
id and systemId / communication between non connected actors

### DISCUSSION THREAD

**Discussion by yanzacheus** [statelyai/xstate#4651] [Q&A] [answered]
Hello
I migrated from v4 to v5, but perhaps misunderstanding some of the improvements.
First I thought that it was now possible to communicate between actors that don't have a parent/child relationship.
So I just created separately my actors, and tried to use sendTo with their id.
1) So I defined their id in the ActorOptions config options parameter of createActor
But it did not work, event if the log said that they started in the right state.
He said to me that I cannot send to the actor from the machine.
2) I tried to duplicate the id (I did not know what to do) in the context of the machine just in case it should be available in the context of the machine. It did not work.
3) I tried to read the docs and see that there was a systemId. So I tried to use it (both in the ActorOptions config options parameter of createActor and the context). The error disappeared, but it was not send to the actor. When I tried to log before system.get('id_of_the_actor'), the result was undefined.

So I don't know what to do: is it finally not possible to send to whatever actor I want ? I thought after that that perhaps, even if the actors don't have to be parent/child to communicate easily, they at least have to be in the "family tree" of an actor ancestor ?

Thanks for any help

**Comment by davidkpiano** [comment, marked as answer]
Can you please share code of what you're trying to do?

Also keep in mind that actors can't magically communicate with any other actor; they can only communicate with actors in the same system (using `systemId`) unless it has a _direct reference_ to the other ActorRef.
<img width="1084" alt="shapes at 24-01-03 13 28 46" src="https://github.com/statelyai/xstate/assets/1093738/a257f7e9-41f0-4568-8f16-6aa0b39e67e7">

**Reply by yanzacheus** [reply]
Ok thanks everything works now thanks to your picture.
This was effectively my question, I needed to spawn/invoke all the actors from an ancestor actor that creates the system so that they can communicate.
Before, I was just using createActor inside a loop for each service, and then trying to make them communicate, but it could'nt work.
 
Just to be sure, another question:
I first set the systemId of my actors with both testid (id of the test) and the serviceid (id of the service in the cloud) like this: `${testId}/${service.id}`. I did this because the tests run in parallel, and I did not want to have systemIds conflicts.  
But from what I understood,  the actors from one test cannot communicate with the actors of another test (they are not in the same system as I recreate a root actor for each test), so I just need to use : systemId: service.id
Is this ok ?
Thanks
